streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java

```diff
@@ -17,7 +17,90 @@
 package org.apache.kafka.streams.processor.internals;
 
 import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.record.TimestampType;
+import org.apache.kafka.streams.errors.DeserializationExceptionHandler;
+import org.apache.kafka.streams.errors.StreamsException;
+import org.apache.kafka.streams.processor.ProcessorContext;
 
-interface RecordDeserializer {
-    ConsumerRecord<Object, Object> deserialize(final ConsumerRecord<byte[], byte[]> rawRecord);
+import static java.lang.String.format;
+import static org.apache.kafka.streams.StreamsConfig.DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG;
+
+class RecordDeserializer {
```

Comment:
Should we add a test for the `RecordDeserializer`?

Reply:
We do have a test. I did compare the "interface" test with "SourceNodeRecordDesrializerTest" and both are exactly the same -- thus, there is not update in `RecordDeserializerTest` in this PR.

Reply:
Sorry if I'm missing something,  but the `SourceNodeRecordDesrializerTest` is removed and I can't see where `RecordDeserializer` is tested.

Reply:
I seems I messed something up here (will add `SourceNodeRecordDesrializerTest` as `RecordDesrializerTest`). Good catch!